### PR TITLE
feat(client): make checkmark a React component for the block

### PR DESCRIPTION
**Pull request by Sboonny**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [ ] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

**Review by huyenltnguyen — APPROVED** [review]

**Comment by Sboonny** [comment]
There were another use case for the checkmark, and it had the same issue https://github.com/freeCodeCamp/freeCodeCamp/pull/50931, the fix is the same for both, so I made it a higher order component and used it, with the JSX syntax.

File: client/src/templates/Introduction/components/challenges.tsx [hunk 1]
```diff
@@ -29,10 +28,6 @@ function Challenges({
   blockTitle
 }: Challenges): JSX.Element {
   const { t } = useTranslation();
-
-  const renderCheckMark = (isCompleted: boolean) =>
-    isCompleted ? <GreenPass /> : <GreenNotCompleted />;
```
Comment (ShaunSHamilton) [inline client/src/templates/Introduction/components/challenges.tsx:34]: This is needed to **not** hush the screen reader text.

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]

File: client/src/templates/Introduction/components/challenges.tsx (hunk 1 above)
Comment (Sboonny) [reply client/src/templates/Introduction/components/challenges.tsx:34]: Yeah, didn't notice, I am going to go with the previous PRs then

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍

**Pull request merged**